util/deephash/deephash.go

```diff
@@ -476,6 +455,59 @@ func (seh sliceElementHasher) hash(h *hasher, p pointer) {
 	}
 }
 
+func makeMapHasher(t reflect.Type) typeHasherFunc {
+	var once sync.Once
+	var hashKey, hashValue typeHasherFunc
+	var isRecursive bool
+	init := func() {
+		hashKey = getTypeInfo(t.Key()).hasher()
+		hashValue = getTypeInfo(t.Elem()).hasher()
+		isRecursive = typeIsRecursive(t)
+	}
+
+	return func(h *hasher, p pointer) {
+		v := p.asValue(t).Elem() // reflect.Map kind
+		if v.IsNil() {
+			h.HashUint8(0) // indicates nil
+			return
+		}
+		once.Do(init)
+		if isRecursive {
+			pm := v.UnsafePointer() // underlying pointer of map
+			if idx, ok := h.visitStack.seen(pm); ok {
+				h.HashUint8(2) // indicates cycle
+				h.HashUint64(uint64(idx))
+				return
+			}
+			h.visitStack.push(pm)
+			defer h.visitStack.pop(pm)
+		}
+		h.HashUint8(1)                // indicates visiting map entries
+		h.HashUint64(uint64(v.Len())) // MUSTDO
```

Comment:
what is MUSTDO?

Reply:
Ooops. It's a reminder to myself to think about whether this is necessary. I couldn't prove to myself that it is. But I couldn't prove to myself that it isn't either. I think it's safer to include.